File: C++/Hard/Accepted/maximum-performance-of-a-team_6.cpp
class Solution {
public:
    int maxPerformance(int n, vector<int>& speed, vector<int>& efficiency, int k) {
        
        vector<pair<int, int>> engineers;
        for (int i = 0; i < n; i++) {
            engineers.push_back({efficiency[i], speed[i]});
        }
        sort(engineers.begin(), engineers.end());
        priority_queue<int, vector<int>, greater<int>> pq;
        long long sum = 0, result = 0;
        for (int i = n - 1; i >= 0; i--) {
            pq.push(engineers[i].second);
            sum += engineers[i].second;
            if (pq.size() > k) {
                sum -= pq.top();
                pq.pop();
            }
            result = max(result, sum * engineers[i].first);
        }
        return result % (long long)(1e9 + 7);
    }
};

File: Rust/Hard/WrongAnswer/maximum-performance-of-a-team_6.rs
impl Solution {
    pub fn max_performance(n: i32, speed: Vec<i32>, efficiency: Vec<i32>, k: i32) -> i32 {
        
        let mut pairs: Vec<(i32, i32)> = speed.iter().zip(efficiency.iter()).map(|(s, e)| (*s, *e)).collect();
        pairs.sort_by_key(|(s, e)| -e);
        let mut heap: std::collections::BinaryHeap<i32> = std::collections::BinaryHeap::new();
        let mut sum = 0;
        let mut res = 0;
        for (s, e) in pairs {
            heap.push(-s);
            sum += s;
            if heap.len() > k as usize {
                sum += heap.pop().unwrap();
            }
            res = res.max(sum * e);
        }
        (res % (1e9 as i32 + 7)) as i32
    }
}